Explorer/Assets/DCL/Friends/UserBlockingCache.cs

```diff
@@ -0,0 +1,85 @@
+using Cysharp.Threading.Tasks;
+using DCL.Friends.UserBlocking;
+using DCL.Web3.Identities;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Utility;
+
+namespace DCL.Friends
+{
+    public class UserBlockingCache : IUserBlockingCache, IDisposable
+    {
+        private readonly IFriendsService friendsService;
+        private readonly IFriendsEventBus eventBus;
+        private readonly IWeb3IdentityCache web3IdentityCache;
+
+        private readonly HashSet<string> blockedUsers = new (StringComparer.OrdinalIgnoreCase);
+        private readonly HashSet<string> blockedByUsers = new (StringComparer.OrdinalIgnoreCase);
+
+        private CancellationTokenSource fetchCts = new ();
+
+        public ReadOnlyHashSet<string> BlockedUsers { get; }
+        public ReadOnlyHashSet<string> BlockedByUsers { get; }
+        public bool HideChatMessages { get; set; }
+
+        public UserBlockingCache(IFriendsService friendsService,
+            IFriendsEventBus eventBus,
+            IWeb3IdentityCache web3IdentityCache)
+        {
+            this.friendsService = friendsService;
+            this.eventBus = eventBus;
+            this.web3IdentityCache = web3IdentityCache;
+
+            BlockedUsers = new ReadOnlyHashSet<string>(blockedUsers);
+            BlockedByUsers = new ReadOnlyHashSet<string>(blockedByUsers);
+
+            eventBus.OnYouBlockedProfile += UserBlockedByYou;
+            eventBus.OnYouUnblockedProfile += UserUnblockedByYou;
+            eventBus.OnYouBlockedByUser += YouBlockedByUser;
+            eventBus.OnYouUnblockedByUser += YouUnblockedByUser;
+
+            web3IdentityCache.OnIdentityChanged += IdentityChanged;
```

Comment:
I dont think the cache itself should manage the identity change. The responsibility should be limited to store data. Instead i think you could subscribe from the friends plugin and make the call to the cache to clear and then refill (similar to what we do with the rpc connection). Then you could decouple `IWeb3IdentityCache` from this class